app/src/main/java/com/greenfox/tribesoflagopusandroid/api/service/MockApiService.java

```diff
@@ -85,7 +86,9 @@ public void enqueue(Callback callback) {
         return new MockCall<Building>() {
             @Override
             public void enqueue(Callback callback) {
-                callback.onResponse(null, Response.success(new Building(1L, type, 1, 1)));
+                Building building = new Building(1L, type, 1, 1);
+                callback.onResponse(null, Response.success(building));
+                buildings.add(building);
```

Comment:
@rekagiday change the order, the "server" saves the new building first, then respond